in_toto/model.go

```diff
@@ -835,8 +843,8 @@ func (mb *Metablock) VerifyWithCertificate(signerKeyID string, rootCertPool *x50
 		return err
 	}
 
-	chain, err := cert.Verify(x509.VerifyOptions{Roots: rootCertPool})
-	if len(chain) == 0 || err != nil {
+	chains, err := cert.Verify(x509.VerifyOptions{Roots: rootCertPool, Intermediates: intermediateCertPool})
```

Comment:
We could build the VerifyOptions earlier and pass it in here to avoid passing around 2 cert pools everywhere.  Unsure how I feel about it.  Refactor work, anyway.

Reply:
Could we allow any number of CAs passed along in an array either in the layout or command line and parse these certificates into a CertPool early. We really have no idea of what the user cert chain will look like.

Reply:
The layout currently does have an array of roots and intermediates, which is where these cert pools are created from.  

I believe the roots & intermediates need to be part of the signed layout.  Otherwise, I could create a root CA at verify time and forge every step with keys signed by my newly created CA.